addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -286,7 +285,16 @@ const Link = Widget.extend({
     _getData: function () {
         var $url = this.$('input[name="url"]');
         var url = $url.val();
-        var content = this.$('input[name="label"]').val() || url;
+        let content = this.data.content;
+        if (this.linkEl) {
+            content = this.linkEl.textContent;
+        }
+        if (this.needLabel) {
+            content = this.el.querySelector('input[name="label"]').value;
+        }
```

Comment:
```suggestion
        if (this.needLabel) {
            content = this.el.querySelector('input[name="label"]').value;
        } else if (this.linkEl) {
            content = this.linkEl.textContent;
        }
```

Reply:
we dont need to check if `this.linkEl` is true if `this.needLabel` is true ;) (I know it's a detail :))

Reply:
Yes that's better, thanks !